block-producer/src/store/mod.rs

```diff
@@ -181,9 +181,20 @@ impl Store for DefaultStore {
             nullifiers.into_iter().collect()
         };
 
-        let tx_inputs = TxInputs {
-            account_hash,
-            nullifiers,
+        let tx_inputs = match account_hash {
+            Some(hash) if hash == Digest::default() => TxInputs {
+                account_hash: None,
+                nullifiers,
+            },
+            Some(hash) => TxInputs {
+                account_hash: Some(hash),
+                nullifiers,
+            },
+            None => {
+                return Err(TxInputsError::MalformedResponse(
+                    "incorrect account hash returned from the store. Got None.".to_string(),
+                ))
+            },
         };
```

Comment:
small nit: I would probably write this as:
```Rust
let account_hash = match account_hash {
    Some(hash) if hash == Digest::default() => None,
    Some(hash) => Some(hash),
    None => {
        return Err(TxInputsError::MalformedResponse(
            "incorrect account hash returned from the store. Got None.".to_string(),
        ))
    },
};

let tx_inputs = TxInputs { account_hash, nullifiers };
```
This logically separates processing of account hash and construction of tx inputs, and makes the code a bit easier to follow.

I would also add a short comment about the assumptions we are making when parsing account hash (i.e., that we assume that default digest means the store does not have the account).